x-pack/filebeat/input/salesforce/input.go

```diff
@@ -341,19 +357,23 @@ func (s *salesforceInput) RunEventLogFile() error {
 
 			resp, err := s.sfdcConfig.Client.Do(req)
 			if err != nil {
-				return err
+				return fmt.Errorf("error fetching log file: %w", err)
 			}
 
-			body, err := io.ReadAll(resp.Body)
-			if err != nil {
+			if resp.StatusCode != http.StatusOK {
 				resp.Body.Close()
```

Comment:
I'm wondering if adding a defer for closing the body would be better here? Since you're closing it here before returning, and again on #L369, `defer` should be able to handle both.

Reply:
Yes defer can be used. But if you see the structure of the function:

- Function is pretty big
- Requests are made for every record
- There are some expensive procedures like CSV unmarshalling, and publishing to ES.

Now, if I use defer the connection will only close when the function returns. That's a huge accumulation in this case i.e., requests are not closed until all CSV are processed as well as published to ES. So, that's why resp.Body.Close is being called at 2 points. Faster closing of requests instead of deferring it to the end.

Reply:
Makes sense.